src/olympia/blocklist/views.py

```diff
@@ -111,15 +125,25 @@ def get_items(apiver, app, appver=None):
     return items, details
 
 
-def get_plugins(apiver, app, appver=None):
+def get_plugins(apiver=3, app=None, appver=None):
     # API versions < 3 ignore targetApplication entries for plugins so only
     # block the plugin if the appver is within the block range.
+
+    if app:
+        app_query = (Q(app__isnull=True) |
+                     Q(app__guid=app) |
+                     Q(app__guid__isnull=True))
+    else:
```

Comment:
I wish I had found a better solution and I hope you can help me find one here.

What we want is to do a ``LEFT OUTER JOIN`` of ``blapps`` so that we can select the extra fields (line 143).
Apparently there are no ``selected_related`` or ``prefetch_related`` similar method in the Django ORM to do the ``LEFT OUTER JOIN`` appart from using filter with ``Q()``

I might be wrong but the Q query I am trying to send is to get all the plugin with an app linked and the one without an app linked.